mozetl/hardware_report/summarize_json.py

```diff
@@ -0,0 +1,716 @@
+"""
+This job was originally located at [1].
+
+[1] https://git.io/vHRWO
+"""
+import ujson as json
+import datetime as dt
+import os.path
+import boto3
+import botocore
+import requests
+import click
+import moztelemetry.standards as moz_std
+
+
+def fetch_json(uri):
+    """ Perform an HTTP GET on the given uri, return the results as json.
+    If there is an error fetching the data, raise an exception.
+
+    Args:
+        uri: the string URI to fetch.
+
+    Returns:
+        A JSON object with the response.
+    """
+    data = requests.get(uri)
+    # Raise an exception if the fetch failed.
+    data.raise_for_status()
+    return data.json()
+
+
+def get_OS_arch(browser_arch, os_name, is_wow64):
+    """ Infers the OS arch from environment data.
+
+    Args:
+        browser_arch: the browser architecture string (either "x86" or "x86-64").
+        os_name: the operating system name.
+        is_wow64: on Windows, indicates if the browser process is running under WOW64.
+
+    Returns:
+        'x86' if the underlying OS is 32bit, 'x86-64' if it's a 64bit OS.
+    """
+
+    is_64bit_browser = browser_arch == 'x86-64'
+    # If it's a 64bit browser build, then we're on a 64bit system.
+    if is_64bit_browser:
+        return 'x86-64'
+
+    is_windows = os_name == 'Windows_NT'
+    # If we're on Windows, with a 32bit browser build, and |isWow64 = true|,
+    # then we're on a 64 bit system.
+    if is_windows and is_wow64:
+        return 'x86-64'
+
+    # Otherwise we're probably on a 32 bit system.
+    return 'x86'
+
+
+def vendor_name_from_id(id):
+    """ Get the string name matching the provided vendor id.
+
+    Args:
+        id: A string containing the vendor id.
+
+    Returns:
+        A string containing the vendor name or "(Other <ID>)" if
+        unknown.
+    """
+
+    # TODO: We need to make this an external resource for easier
+    # future updates.
+    vendor_map = {
+        '0x1013': 'Cirrus Logic',
+        '0x1002': 'AMD',
+        '0x8086': 'Intel',
+        '0x5333': 'S3 Graphics',
+        '0x1039': 'SIS',
+        '0x1106': 'VIA',
+        '0x10de': 'NVIDIA',
+        '0x102b': 'Matrox',
+        '0x15ad': 'VMWare',
+        '0x80ee': 'Oracle VirtualBox',
+        '0x1414': 'Microsoft Basic',
+    }
+
+    return vendor_map.get(id, "Other")
+
+
+def get_device_family_chipset(vendor_id, device_id):
+    """ Get the family and chipset strings given the vendor and device ids.
+
+    Args:
+        vendor_id: a string representing the vendor id (e.g. '0xabcd').
+        device_id: a string representing the device id (e.g. '0xbcde').
+
+    Returns:
+        A string in the format "Device Family Name-Chipset Name".
+    """
+    if not vendor_id in device_map:
```

Comment:
My guess is this is causing the tests to fail. Can we send in device_map as a param to make this a fully functional function?